lightning/src/ln/channel.rs

```diff
@@ -1350,12 +1360,63 @@ impl<SP: Deref> Channel<SP> where
 				}
 			},
 			#[cfg(not(dual_funding))]
-			Channel::UnfundedV2(_) => {
+			ChannelPhase::UnfundedV2(_) => {
 				debug_assert!(false);
 				Ok(None)
 			},
 		}
 	}
+
+	pub fn funding_signed<L: Deref>(
+		&mut self, msg: &msgs::FundingSigned, best_block: BestBlock, signer_provider: &SP, logger: &L
+	) -> Result<(&mut FundedChannel<SP>, ChannelMonitor<<SP::Target as SignerProvider>::EcdsaSigner>), ChannelError>
+	where
+		L::Target: Logger
+	{
+		let phase = core::mem::replace(&mut self.phase, ChannelPhase::Intermediate);
```

Comment:
The `Intermediate` is a bit scary with all the `unreachable()`s. I would like to see a comment here what's going on -- that it will be set to something else in this function, or even more, a debug assert at the end of the function to make sure it is set to something else.
BTW, it's a nice trick to be able to take out an inner part of the phase and put it back.

Reply:
Good idea. Added `debug_assert`s.